### PR TITLE
[DCP][nit] Rename variables + minor documentation fix for optimizer.py

### PR DESCRIPTION
**Pull request by wz337**

### CODE REVIEW

File: torch/distributed/checkpoint/optimizer.py [hunk 1]
```diff
@@ -202,15 +202,15 @@ def load_sharded_optimizer_state_dict(
     storage_reader: dist_cp.StorageReader,
 ) -> STATE_DICT_TYPE:
     """
-    Loads a state_dict to be used in conjuntion with FSDP sharded optimizer state.
-    This is the current recommended way to checkpoint is FSDP
+    Loads a state_dict in conjuntion with FSDP sharded optimizer state.
```
Comment (rohan-varma) [inline torch/distributed/checkpoint/optimizer.py:205]: just for my learning, is this function specific to FSDP?

**Review by rohan-varma — APPROVED** [review]
LGMT

File: torch/distributed/checkpoint/optimizer.py (hunk 1 above)
Comment (wz337) [reply torch/distributed/checkpoint/optimizer.py:205]: Yes!

**Pull request closed without merging**